aten/src/ATen/native/Math.h

```diff
@@ -381,6 +381,455 @@ static inline float calc_polygamma(int64_t n, float x) {
       zeta(double(n + 1), x);
 }
 
+// regularized lower incomplete gamma
+// the regularized lower and upper incomplete gamma follows SciPy's
```

Comment:
Wait is there going to be an implementation of the regularized upper incomplete gamma function, too? I thought this PR was just the regularized lower incomplete gamma function.

Reply:
The upper incomplete gamma function is required to calculate the lower incomplete gamma function, so I just add it here, but I don't add the upper one as an API.

Reply:
That makes mathematical sense, of course.

As an aside, I don't think I like either the name "igamma" or "gammainc." Both suggest this is the "incomplete Gamma function," which isn't even a thing, and definitely not the regularized lower/upper incomplete Gamma function. SciPy's and Tensorflow's names for the regularized upper incomplete Gamma function seem even sillier. Oh well.

My question then, should this PR also add igammac? Or would you be interested in adding it in a follow-up PR?

Reply:
I'd prefer to finish this first before jumping into `igammac`. It's giving me quite plenty things to learn already.